unpack-trees.c

```diff
@@ -1437,7 +1437,9 @@ static void mark_new_skip_worktree(struct exclude_list *el,
 	 * 2. Widen worktree according to sparse-checkout file.
 	 * Matched entries will have skip_wt_flag cleared (i.e. "in")
 	 */
+	enable_fscache(0);
```

Comment:
It might make sense to initialize the hashmap size of the FSCache with `istate->cache_nr`? That way, the hashmap does not have to grow a large number of times before reaching its final size.

Reply:
This appears to have improved the performance to ~6 seconds!

Reply:
Wow! :+1: